armory/paths.py

```diff
@@ -9,58 +9,123 @@
-
-    # TODO: Recommend to use armory config (#165)
-    except KeyError:
-        raise KeyError(
-            "Error parsing config.json. If you previously ran an older "
-            "version of armory you may need to remove the ~/.armory directory since"
-            "permission changes."
+        validate_config(config)
+    except (TypeError, KeyError, ValueError):
+        logger.error(
+            "Error parsing config.json. Please run `armory configure`.\n"
+            "    If you previously ran an older version of armory, you may\n"
+            f"    need to remove the {default().armory_dir} directory due to changes"
         )
+        raise
+
     return config
 
 
 class HostPaths:
+    def __init__(self):
+        self.cwd = default().cwd
+        self.user_dir = default().user_dir
+        self.armory_dir = default().armory_dir
+        self.armory_config = default().armory_config
+        if os.path.isfile(self.armory_config):
+            # Parse paths from config
+            config = load_config()
+            for k in "dataset_dir", "saved_model_dir", "output_dir", "tmp_dir":
+                setattr(self, k, config[k])
+            self.external_repo_dir = os.path.join(self.tmp_dir, "external")
+        else:
+            logger.warning(f"No {self.armory_config} file. Using default paths.")
+            logger.warning("Please run `armory configure`")
+            self.dataset_dir = default().dataset_dir
+            self.saved_model_dir = default().saved_model_dir
+            self.tmp_dir = default().tmp_dir
+            self.output_dir = default().output_dir
+            self.external_repo_dir = default().external_repo_dir
+
+    def makedirs(self):
+        logger.info("Creating armory directories if they do not exist")
+        os.makedirs(self.dataset_dir, exist_ok=True)
+        os.makedirs(self.saved_model_dir, exist_ok=True)
+        os.makedirs(self.tmp_dir, exist_ok=True)
+        os.makedirs(self.output_dir, exist_ok=True)
+
+
+class HostDefault:
     def __init__(self):
         self.cwd = os.getcwd()
         self.user_dir = os.path.expanduser("~")
         self.armory_dir = os.path.join(self.user_dir, ".armory")
-        self.tmp_dir = os.path.join(self.armory_dir, "tmp")
         self.armory_config = os.path.join(self.armory_dir, "config.json")
 
-        # Parse paths from config
-        config = _parse_global_config(self.armory_config, self.armory_dir)
-        self.dataset_dir = config.get("cached_dataset_dir")
-        self.output_dir = config.get("output_dir")
-        self.external_repo_dir = os.path.join(self.dataset_dir, "external_repos")
+        self.dataset_dir = os.path.join(self.armory_dir, "datasets")
+        self.saved_model_dir = os.path.join(self.armory_dir, "saved_models")
+        self.tmp_dir = os.path.join(self.armory_dir, "tmp")
+        self.output_dir = os.path.join(self.armory_dir, "outputs")
+        self.external_repo_dir = os.path.join(self.dataset_dir, "external")
```

Comment:
tmp dir instead of dataset? Getting:
```
2020-02-28 12:55:42 ubuntu-spmorgan armory.paths[2941] INFO Creating armory directories if they do not exist
2020-02-28 12:55:43 ubuntu-spmorgan root[2941] INFO Downloading external repo: twosixlabs/armory-example
2020-02-28 12:55:43 ubuntu-spmorgan armory.docker.management[2941] INFO ARMORY Instance 28011266d5 created.
2020-02-28 12:55:43 ubuntu-spmorgan armory.eval.evaluator[2941] INFO Running evaluation script
/opt/conda/bin/python: Error while finding module specification for 'armory-example.evaluations.fgm_attack' (ModuleNotFoundError: No module named 'armory-example')

2020-02-28 12:55:44 ubuntu-spmorgan armory.eval.evaluator[2941] WARNING Shutting down container
2020-02-28 12:55:44 ubuntu-spmorgan armory.docker.management[2941] INFO Stopping instance: 28011266d5
```

Reply:
Good catch on that. What are you running for test?

Reply:
I think the error you are getting is different from the line 120 issue.

Reply:
Was your output using the host default? I.e., were you getting the warnings:
```
(armory) davidslater@aleph-5:~/git/davidslater/armory$ python -m armory run tests/test_data/fgm_attack_test.json 
2020-02-28 10:33:20 aleph-5.local armory.paths[96750] WARNING No /Users/davidslater/.armory/config.json file. Using default paths.
2020-02-28 10:33:20 aleph-5.local armory.paths[96750] WARNING Please run `armory configure`
2020-02-28 10:33:20 aleph-5.local armory.paths[96750] INFO Creating armory directories if they do not exist
```

Reply:
I've pushed that change.